pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/IntegerTupleSketchAggregationFunction.java

```diff
@@ -0,0 +1,213 @@
+      byte[][] bytesValues = blockValSet.getBytesValuesSV();
+      try {
+        for (int i = 0; i < length; i++) {
+          byte[] value = bytesValues[i];
+          int groupKey = groupKeyArray[i];
+          CompactSketch<IntegerSummary> newSketch =
+              ObjectSerDeUtils.DATA_SKETCH_INT_TUPLE_SER_DE.deserialize(value).compact();
+          if (groupByResultHolder.getResult(groupKey) == null) {
+            ArrayList<CompactSketch<IntegerSummary>> newList = new ArrayList<>();
+            newList.add(newSketch);
+            groupByResultHolder.setValueForKey(groupKey, newList);
+          } else {
+            groupByResultHolder.<List<CompactSketch<IntegerSummary>>>getResult(groupKey).add(newSketch);
+          }
+        }
+      } catch (Exception e) {
+        throw new RuntimeException("Caught exception while merging Tuple Sketches", e);
+      }
+    } else {
+      throw new IllegalStateException(
+          "Illegal data type for INTEGER_TUPLE_SKETCH_UNION aggregation function: " + storedType);
+    }
+  }
+
+  @Override
+  public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] valueArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    for (int i = 0; i < length; i++) {
+      byte[] value = valueArray[i];
+      CompactSketch<IntegerSummary> newSketch =
+          ObjectSerDeUtils.DATA_SKETCH_INT_TUPLE_SER_DE.deserialize(value).compact();
+      for (int groupKey : groupKeysArray[i]) {
+        if (groupByResultHolder.getResult(groupKey) == null) {
+          groupByResultHolder.setValueForKey(groupKey, Collections.singletonList(newSketch));
+        } else {
+          groupByResultHolder.<List<CompactSketch<IntegerSummary>>>getResult(groupKey).add(newSketch);
+        }
+      }
+    }
+  }
+
+  @Override
+  public List<CompactSketch<IntegerSummary>> extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    return aggregationResultHolder.getResult();
+  }
+
+  @Override
+  public List<CompactSketch<IntegerSummary>> extractGroupByResult(GroupByResultHolder groupByResultHolder,
+      int groupKey) {
+    return groupByResultHolder.getResult(groupKey);
+  }
+
+  @Override
+  public List<CompactSketch<IntegerSummary>> merge(List<CompactSketch<IntegerSummary>> intermediateResult1,
+      List<CompactSketch<IntegerSummary>> intermediateResult2) {
+    if (intermediateResult1 == null && intermediateResult2 != null) {
+      return intermediateResult2;
+    } else if (intermediateResult1 != null && intermediateResult2 == null) {
+      return intermediateResult1;
+    } else if (intermediateResult1 == null && intermediateResult2 == null) {
+      return new ArrayList<>(0);
+    }
+    ArrayList<CompactSketch<IntegerSummary>> merged =
+        new ArrayList<>(intermediateResult1.size() + intermediateResult2.size());
+    merged.addAll(intermediateResult1);
```

Comment:
Just curious - We dont want to do a union here for the merge?  Im looking at DistinctCountThetaSketchAggregationFunction for reference.

Reply:
this is an optimisation similar to the one used in the Theta version, where merges can be quite expensive, and it's better to delay the merge til we have a lot of sketches to combine, hence using `List` as the intermediate type